showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITHttpAnnotation.java

```diff
@@ -0,0 +1,104 @@
+    ComplianceSuite.Builder builder = ComplianceSuite.newBuilder();
+    JsonFormat.parser()
+        .merge(
+            new InputStreamReader(
+                ITHttpAnnotation.class
+                    .getClassLoader()
+                    .getResourceAsStream("compliance_suite.json")),
+            builder);
+    complianceSuite = builder.build();
+
+    ComplianceSettings httpjsonComplianceSettings =
+        ComplianceSettings.newHttpJsonBuilder()
+            .setCredentialsProvider(NoCredentialsProvider.create())
+            .setTransportChannelProvider(
+                ComplianceSettings.defaultHttpJsonTransportProviderBuilder()
+                    .setHttpTransport(
+                        new NetHttpTransport.Builder().doNotValidateCertificate().build())
+                    .setEndpoint("http://localhost:7469")
+                    .build())
+            .build();
+    complianceClient = ComplianceClient.create(httpjsonComplianceSettings);
+
+    // Mapping of Compliance Suite file RPC Names to ComplianceClient methods
+    complianceValidRpcMap =
+        ImmutableMap.of(
+            "Compliance.RepeatDataBody",
+            complianceClient::repeatDataBody,
+            "Compliance.RepeatDataBodyInfo",
+            complianceClient::repeatDataBodyInfo,
+            "Compliance.RepeatDataQuery",
+            complianceClient::repeatDataQuery,
+            "Compliance.RepeatDataSimplePath",
+            complianceClient::repeatDataSimplePath,
+            "Compliance.RepeatDataBodyPut",
+            complianceClient::repeatDataBodyPut,
+            "Compliance.RepeatDataBodyPatch",
+            complianceClient::repeatDataBodyPatch,
+            "Compliance.RepeatDataPathResource",
+            complianceClient::repeatDataPathResource);
+  }
+
+  @After
+  public void destroyClient() {
+    complianceClient.close();
+  }
+
+  /*
+  This test loops through all the testing groups (which test different aspects of the http annotation)
+  and calls the listed RPC Names (which are the different HttpVerbs) with the given request. The response's
+  info should be the same as the request's. This tests that the request is set correctly into the corresponding
+  URL/Body/Query.
+   */
+  @Test
+  public void testHttpAnnotations() {
+    for (ComplianceGroup compliancegroup : complianceSuite.getGroupList()) {
+      List<String> validRpcList =
+          compliancegroup.getRpcsList().stream()
+              .filter(complianceValidRpcMap::containsKey)
+              .collect(Collectors.toList());
+      for (String rpcName : validRpcList) {
+        Function<RepeatRequest, RepeatResponse> rpc = complianceValidRpcMap.get(rpcName);
+        for (RepeatRequest repeatRequest : compliancegroup.getRequestsList()) {
+          ComplianceData expectedData = repeatRequest.getInfo();
+          RepeatResponse response = rpc.apply(repeatRequest);
+          assertThat(response.getRequest().getInfo()).isEqualTo(expectedData);
```

Comment:
This may require a big change but was wondering - would it be possible to explicitly specify the expected value is without computing it?

Reply:
As written, the compliance_suite.json is defined in a language-neutral way and can be (should?*) be owned by the gapic-showcase repo. If we hardcode expected values in the java test that duplicates the expected input, I would think this would eliminate our ability to have a common test suite defined elsewhere.

(*) Is there a way to get the compliance_suite.json from the gapic-showcase dependency rather than copying it into this repo?

Reply:
Great - this is perfect.

I vote to .gitignore it since it's a "generated" file to avoid confusion over the source of truth; but the main concern you've already covered.

Thank you.

Reply:
Sounds good. Going to loop in @blakeli0 again if he has some other concerns that might be missed. If not, I can add the file back into the .gitignore

Reply:
re comment on hard-coding the expected value: Thanks for the clarification! I didn't take this aspect into account: `I would think this would eliminate our ability to have a common test suite defined elsewhere.` Given this reasoning, computing the value instead of hard-cording looks like the way to go.

Reply:
> I vote to .gitignore it since it's a "generated" file to avoid confusion over the source of truth; but the main concern you've already covered.

I feel it might be better to include this `compliance_suite.json` file in our repo and maybe the protos as well, as it would be easier to read and troubleshoot the tests. Even though they are "generated", but I consider them as part of the whole gapic-showcase test suite, just like the "generated" gapic client libraries, which are already included in our repo.

Reply:
Sure, added the json file in.